### PR TITLE
Add ActionController::API support to Stronger Parameters. 

### PR DESCRIPTION
**Pull request by swanjr**
Description

This allows StongerParameters use with API only rails controllers which currently does not work.
Added test for new code.

### CODE REVIEW

File: lib/stronger_parameters/parameters.rb [hunk 1]
```diff
@@ -194,7 +195,22 @@ module ControllerSupport
...
+      rescue_from(StrongerParameters::InvalidParameter) do |e|
+        render json: { error: e.message }, status: :bad_request # uncovered
+      end
+    end
+  end
 end
 
 ActionController::Parameters.include StrongerParameters::Parameters
 ActionController::Base.include StrongerParameters::ControllerSupport
+ActionController::API.include StrongerParameters::APIControllerSupport
```
Comment (grosser) [inline lib/stronger_parameters/parameters.rb:216]: because base is not in api right ?

**Review by grosser — COMMENTED** [review]
it should include the same module into both and check if it's json or not
... the base controller should also show json when a json request fails

File: lib/stronger_parameters/parameters.rb (hunk 1 above)
Comment (swanjr) [reply lib/stronger_parameters/parameters.rb:216]: Correct, API is a stripped down version of Base.

File: lib/stronger_parameters/parameters.rb [hunk 2]
```diff
@@ -190,11 +191,16 @@ module ControllerSupport
       # TODO: this is not consistent with the behavior of raising ActionController::UnpermittedParameters
       # should have the same render vs raise behavior in test/dev ... see permitted_parameters_test.rb
       rescue_from(StrongerParameters::InvalidParameter) do |e|
-        render plain: e.message, status: :bad_request # uncovered
+        if request.format.to_s.include?('json')
```
Comment (swanjr) [inline lib/stronger_parameters/parameters.rb:194]: FYI avoided use to `respond_to do |format|` since it is included in ActionController::API by default.

**Review by grosser — APPROVED** [review]

**Comment by grosser** [comment]
2.17.0

**Pull request merged**